aptos-move/framework/aptos-stdlib/sources/type_info.move

```diff
@@ -22,6 +24,32 @@ module aptos_std::type_info {
     public native fun type_of<T>(): TypeInfo;
     public native fun type_name<T>(): string::String;
 
+    /// Return the size, in bytes, of a `T`, provided an immutable
+    /// reference to a quasi-null instance of fixed-size `T`,
+    /// `fixed_size_type_null_ref`.
+    ///
+    /// Analogous to `sizeof()` in C.
+    ///
+    /// Ideally this function would be implemented as a native function
+    /// of the form `public native fun size_of<T>(): u64;`, such that
+    /// callers do not need to concoct quasi-null instances, e.g.
+    /// `false` for `T` as a `bool`, `0` for `T` as a `u8`, or `@0x0`
+    /// for `T` as an `address`.
+    ///
+    /// Does not actually enfore that `T` is a fixed-size type, which
+    /// would require determining whether or not vectors are nested
+    /// within.
+    ///
+    /// See `test_size_of()` for an analysis of common types and nesting
+    /// patterns, as well as `test_size_of_vectors()` for an analysis of
+    /// vector size dynamism.
+    public fun size_of<T>(
```

Comment:
it's called `size_of_val` in rust, `size_of` typically calls on type instead of values. I also think it's probably better a native than a bcs serialization. @wrwg thoughts?